include/swift/AST/DiagnosticsSema.def

```diff
@@ -4472,13 +4472,19 @@ ERROR(differentiable_function_type_invalid_parameter,none,
       "%select{| and satisfy '%0 == %0.TangentVector'}1, but the enclosing "
       "function type is '@differentiable%select{|(linear)}1'"
       "%select{|; did you want to add '@noDerivative' to this parameter?}2",
-      (StringRef, /*tangentVectorEqualsSelf*/ bool,
+      (StringRef, /*isLinear*/ bool,
        /*hasValidDifferentiabilityParameter*/ bool))
 ERROR(differentiable_function_type_invalid_result,none,
       "result type '%0' does not conform to 'Differentiable'"
       "%select{| and satisfy '%0 == %0.TangentVector'}1, but the enclosing "
       "function type is '@differentiable%select{|(linear)}1'",
       (StringRef, bool))
+ERROR(differentiable_function_type_missing_differentiability_parameter, 
```

Comment:
No strong preference, but maybe this name is slightly clearer:

```suggestion
ERROR(differentiable_function_type_no_differentiability_parameters, 
```

Reply:
Good suggestion. Done.